packages/proxy/lib/http/response-middleware.ts

```diff
@@ -393,99 +393,99 @@ const MaybePreventCaching: ResponseMiddleware = function () {
   this.next()
 }
 
-const determineIfNeedsCrossOriginHandling = (ctx: HttpMiddlewareThis<ResponseMiddlewareProps>) => {
-  const previousAUTRequestUrl = ctx.getPreviousAUTRequestUrl()
+const checkNeedsCrossOriginHandling = (ctx: HttpMiddlewareThis<ResponseMiddlewareProps>) => {
+  const currentAUTUrl = ctx.getAUTUrl()
 
   // A cookie needs cross origin handling if it's an AUT request and
   // either the request itself is cross-origin or the origins between
   // requests don't match, since the browser won't set them in that
   // case and if it's secondary-origin -> primary-origin, we don't
   // recognize the request as cross-origin
   return (
-    !!ctx.req.isAUTFrame &&
-    (
-      (previousAUTRequestUrl && !cors.urlOriginsMatch(previousAUTRequestUrl, ctx.req.proxiedUrl))
+    ctx.config.experimentalSessionAndOrigin
+    && ctx.req.isAUTFrame
+    && (
+      (currentAUTUrl && !cors.urlOriginsMatch(currentAUTUrl, ctx.req.proxiedUrl))
       || !ctx.remoteStates.isPrimaryOrigin(ctx.req.proxiedUrl)
     )
   )
 }
 
-interface EnsureSameSiteNoneProps {
-  cookie: string
-  browser: Browser | { family: string | null }
-  isLocalhost: boolean
-  url: URL
-  ctxDebug: Debug.Debugger
-}
-
-const cookieSameSiteRegex = /SameSite=(\w+)/i
-const cookieSecureRegex = /(^|\W)Secure(\W|$)/i
-const cookieSecureSemicolonRegex = /;\s*Secure/i
-
-const ensureSameSiteNone = ({ cookie, browser, isLocalhost, url, ctxDebug }: EnsureSameSiteNoneProps) => {
-  ctxDebug('original cookie: %s', cookie)
+const CopyCookiesFromIncomingRes: ResponseMiddleware = async function () {
+  const cookies: string | string[] | undefined = this.incomingRes.headers['set-cookie']
```

Comment:
in what case is this a single string vs an array of strings?

Reply:
Honestly, I don't know. I'm just going off the prior art for this header.

Reply:
I would guess it may be a string if there's a single `Set-Cookie` header and it's an array if there's more than one.

Reply:
That makes sense. I was wondering if the code would accidentally treat the string as a character array, but it looks like it handles it just as well if `cookies` happened to be a string.

Reply:
That's where the `[].concat(cookies)` idiom comes in to make sure it's an array before we loop through it.